File: cpp/Stack/easy/build-an-array-with-stack-operations.cpp
class Solution {
public:
    vector<string> buildArray(vector<int>& target, int n) {
        vector<string> str;
        
        int i = 0;
        int j = 1;
        while(i < target.size()) {
            if(target[i] == j) {
                str.push_back("Push");
                i++;
                j++;
            } else {
                str.push_back("Push");
                str.push_back("Pop");
                j++;
            }
        }
        return str;
    }
};

File: java/Stack/easy/build-an-array-with-stack-operations.java
class Solution {
    public List<String> buildArray(int[] target, int n) {
        List<String> a=new ArrayList<String> ();
        int t=0;
        for(int i=1;i<=n && t<target.length;i++){
            if (target[t]==i){
                t++;
                a.add("Push");
            }
            else{
                a.add("Push");
                a.add("Pop");
            }
        }return a;
    }
}
